runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1666,13 +1666,78 @@ TR_VectorAPIExpansion::transformIL(bool checkBoxing)
          continue;
 
       bool scalarized;
-      bool vectorizedNode;
-
+      bool vectorizedOrScalarizedNode;
       TR::DataType elementType;
       int32_t bitsLength;
       vapiObjType objectType;
 
-      bool vectorizedOrScalarizedNode = isVectorizedOrScalarizedNode(node, elementType, bitsLength, objectType, scalarized);
+      vectorizedOrScalarizedNode = isVectorizedOrScalarizedNode(node, elementType, bitsLength, objectType, scalarized);
+
+      // Vectorize intrinsic if its operands are known
+      if (boxingAllowed() &&
+         !vectorizedOrScalarizedNode &&
+         opCode.isFunctionCall())
+         {
+         TR::MethodSymbol * methodSymbol = node->getSymbolReference()->getSymbol()->castToMethodSymbol();
+         if (isVectorAPIMethod(methodSymbol))
+             {
+             if (methodSymbol->getRecognizedMethod() == TR::jdk_internal_vm_vector_VectorSupport_compare)
+                {
+                for (int i = 0; i < 2; i++)
+                   {
+                   TR::Node* operand = node->getChild(getFirstOperandIndex(methodSymbol) + i);
+
+                   bool operandVectorizedOrScalarized;
```

Comment:
You can declare and initialize the variable in a single line by moving this declaration to L.1696.